pkg/deploy/lattice/rule_manager.go

```diff
@@ -200,6 +180,9 @@ func (r *defaultRuleManager) Upsert(
 	}
 	// TODO: fetching all rules every time is not efficient - maybe have a separate public method to prepopulate?
 	currentLatticeRules, err := r.cloud.Lattice().GetRulesAsList(ctx, &lri)
+	if err != nil {
+		return model.RuleStatus{}, fmt.Errorf("failed to GetRulesAsList, err: %s", err)
```

Comment:
1. Better to use %w for error wrap.
2. When you **wrap** errors it's better not to use words "failed". If every wrapper tend to adds them final error afte will have too many "failed"s

Result looks like this:
`"reconciler failed: manager failed: failed to upsert: failed to GetRules: lattice 400 error"`

But should be like this:
`"reconciler: manager: upsert: GetRules: lattice 400 error"`